### PR TITLE
adds plate carrier armor, adds accessories being taken into account for armor, properly implements armor_absorb this time.

### PR DESCRIPTION
**Pull request by MadmanMartian**
Plate carrier armor, sprites and idea courtesy of @TheBimmer231

Plate carrier armor doesn't have any actual armor value, but accepts armor plates that actually use armor_absorb to reduce damage. As they take punishment, they take damage until they eventually fall apart.

Still testing the numbers, somewhat hard after finding #17416

Also adds armor shards you can attach to your jumpsuit (Shoulder pads, knee pads, face guard) to give you a bit more protection.

Also adds clothing-level procs for get armor and get armor absorb to be overridden

###TODO
- [x] Add method of acquisition
- [x] Finish testing
- [x] Balance

:cl:
* rscadd: Adds ablative plate armor
* rscadd: Adds armor shards you can attach to your jumpsuit to give you a bit more armor

### CODE REVIEW

**Comment by Lokiamis** [comment]
Pictures

**Comment by TheBimmer231** [comment]
@Lokiamis 
image
how it should look at the end.

**Comment by Lokiamis** [comment]
My concern is that there’s no way for someone to tell if they have a plate that defends against lasers or bullets

**Comment by SonixApache** [comment]
Dunno if you know, but you can just use the guest function to local-test with yourself.
Or, well, you don't even need to, you can just dress some guy up and shoot him, right?
What's the difficulty with testing, exactly?

**Comment by MadmanMartian** [comment]
This bot is so fucking handy, it just caught that I accidentally removed the new bone sheets

File: code/game/objects/effects/effect_system.dm [hunk 1]
```diff
@@ -446,7 +446,8 @@ steam.start() -- spawns the effect
 			n = 20
 		number = n
 		cardinals = c
-		carry.copy_to(chemholder, carry.total_volume)
```
Comment (Probe1) [inline code/game/objects/effects/effect_system.dm]: [NONATOMIC]

File: code/modules/clothing/suits/plate_carrier.dm [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+/obj/item/clothing/suit/armor/plate_carrier/attack_self(mob/user)
+	if(P)
+		user.put_in_hands(P)
+		P = null
+
+/obj/item/clothing/suit/armor/plate_carrier/attackby(obj/item/W,mob/user)
+	..()
+	if(istype(W, /obj/item/weapon/armor_plate) && !P && user.drop_item(W, src))
+		P = W
+		to_chat(user, "<span class = 'notice'>You install \the [W] into \the [src].</span>")
```
Comment (Probe1) [inline code/modules/clothing/suits/plate_carrier.dm]: It needs to output to chat when there's already a plate in there.

File: code/modules/clothing/suits/plate_carrier.dm [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+	name = "ceramic armor plate"
+	health = 20
+	armor = list(melee = 25, bullet = 7, laser = 50, energy = 10, bomb = 25, bio = 0, rad = 0)
+	armor_absorb = list(melee = 25, bullet = 20, laser = 20, energy = -5, bomb = 0, bio = 0, rad = 0)
+
+
+/obj/item/weapon/armor_plate/proc/receive_damage(var/type, var/amount)
+	if(type == BRUTE || type == BURN)
+		health -= amount
+	playsound(get_turf(src), 'sound/effects/Glasshit.ogg', 70, 1)
```
Comment (Probe1) [inline code/modules/clothing/suits/plate_carrier.dm:87]: I was able to get this to loop.  The mob was knocked over by ZAS and slammed into objects.  The plate was destroyed but the sound kept looping until I took the armor off the mob.  I'm not sure what this indicates.

File: code/modules/clothing/suits/plate_carrier.dm [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+		if(initial(health)/2-1 to initial(health)/4)
+			to_chat(user, "<span class = 'warning'>\The [src] is brittle.</span>")
+		if(initial(health)/4-1 to 0)
+			to_chat(user, "<span class = 'warning'>\The [src] is falling apart!</span>")
+
+/obj/item/weapon/armor_plate/bullet_resistant
+	name = "plasteel armor plate"
+	icon_state = "plate_2"
+	health = 30
+	armor = list(melee = 50, bullet = 80, laser = 10, energy = 10, bomb = 0, bio = 0, rad = 0)
```
Comment (Probe1) [inline code/modules/clothing/suits/plate_carrier.dm]: Dial this down to 50 bullet.  As stands you can take all 8 buckshot rounds from a combat shotgun and it still doesn't break.  That's going to feel like utter shit for players when someone has super armor that's as easy to get as a stop to cargo and opening a crate.

**Review by Probe1 — COMMENTED** [review]
Code looks fine but there's a QoL and balance change to be made.

**Comment by Probe1** [comment]
How are the items in ``` code/modules/clothing/accessories/armor_shards.dm ``` created?

**Comment by DamianX** [comment]
when

**Comment by MadmanMartian** [comment]
should be good to go

File: code/modules/clothing/head/helmet.dm [hunk 5]
```diff
@@ -10,6 +10,48 @@
...
+			armor = initial(armor)
+			icon_state = "[initial(icon_state)]_down"
+			item_state = "[initial(item_state)]_down"
+		if(0)
+			armor = list(melee = 50, bullet = 10, laser = 50, energy = 10, bomb = 25, bio = 0, rad = 0)
+			icon_state = "[initial(icon_state)]_up"
+			item_state = "[initial(item_state)]_down"
+	if(user)
+		user.update_inv_head()
+		usr.update_inv_wear_mask()
```
Comment (DamianX) [inline code/modules/clothing/head/helmet.dm]: usr?

**Review by DamianX — DISMISSED** [review]

**Comment by MadmanMartian** [comment]
when?

**Comment by ghost** [comment]
If people feel it's a balance issue for the opponent identifying the armor type like all the other armors, change the pouches to be color coded and overlay with either black for the ballistic plate carrier or blue for the energy plate carrier?

**Comment by Probe1** [comment]
Sure

**Pull request merged**